From 52bfb2f049b534aca5b6d3e7475c9b2dd97c55a3 Mon Sep 17 00:00:00 2001
From: Johannes Demel <demel@uni-bremen.de>
Date: Tue, 17 Mar 2020 21:20:51 +0100
Subject: [PATCH 2/7] exp: Rename SSE4.1 to SSE2 kernel

The SSE kernel only requires SSE2 instructions. Thus, we can just use
this instruction level.
---
 kernels/volk/volk_32f_exp_32f.h | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/kernels/volk/volk_32f_exp_32f.h b/kernels/volk/volk_32f_exp_32f.h
index 19c3d9d..26fdf02 100644
--- a/kernels/volk/volk_32f_exp_32f.h
+++ b/kernels/volk/volk_32f_exp_32f.h
@@ -99,11 +99,11 @@
 #ifndef INCLUDED_volk_32f_exp_32f_a_H
 #define INCLUDED_volk_32f_exp_32f_a_H
 
-#ifdef LV_HAVE_SSE4_1
-#include <smmintrin.h>
+#ifdef LV_HAVE_SSE2
+#include <emmintrin.h>
 
 static inline void
-volk_32f_exp_32f_a_sse4_1(float* bVector, const float* aVector, unsigned int num_points)
+volk_32f_exp_32f_a_sse2(float* bVector, const float* aVector, unsigned int num_points)
 {
   float* bPtr = bVector;
   const float* aPtr = aVector;
@@ -175,7 +175,7 @@ volk_32f_exp_32f_a_sse4_1(float* bVector, const float* aVector, unsigned int num
   }
 }
 
-#endif /* LV_HAVE_SSE4_1 for aligned */
+#endif /* LV_HAVE_SSE2 for aligned */
 
 
 #ifdef LV_HAVE_GENERIC
@@ -199,11 +199,11 @@ volk_32f_exp_32f_a_generic(float* bVector, const float* aVector, unsigned int nu
 #ifndef INCLUDED_volk_32f_exp_32f_u_H
 #define INCLUDED_volk_32f_exp_32f_u_H
 
-#ifdef LV_HAVE_SSE4_1
-#include <smmintrin.h>
+#ifdef LV_HAVE_SSE2
+#include <emmintrin.h>
 
 static inline void
-volk_32f_exp_32f_u_sse4_1(float* bVector, const float* aVector, unsigned int num_points)
+volk_32f_exp_32f_u_sse2(float* bVector, const float* aVector, unsigned int num_points)
 {
   float* bPtr = bVector;
   const float* aPtr = aVector;
@@ -276,7 +276,7 @@ volk_32f_exp_32f_u_sse4_1(float* bVector, const float* aVector, unsigned int num
   }
 }
 
-#endif /* LV_HAVE_SSE4_1 for unaligned */
+#endif /* LV_HAVE_SSE2 for unaligned */
 
 
 #ifdef LV_HAVE_GENERIC
-- 
2.20.1

